### PR TITLE
add products from shop to cart, tests.

### LINKED ISSUE #58: add item in shop to cart
Have a "add to cart" Button for each product in shop.

### PR DESCRIPTION
**Pull request by Rebecca-Elliott**
## PR type

- [X] Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Performance Optimization
- [ ] Documentation Update

## Description

Adds the ability to add products from the shop to the cart.
Added additional entities to the demo for testing.

## Related Issues/Tickets

closes #58 

## QA Instruction, Screenshots etc...

uncomment the demo in AccessingDataJpaApplication.java.
run app. go to http://localhost:8080/home/buyer/2. From there you can search for a shop and look at it's products. Shop "U" is preloaded with items. From the shop items can be added to the cart. To navigate to the cart click on the icon labeled "My Orders".

https://github.com/user-attachments/assets/e398cd8e-d506-4f08-8823-d58e30f237eb

## Added/updated tests?

- [X ] Yes
- [ ] No: _replace this line with details on why tests are not included in this PR_

## PR Checklist

- [X] I have rebased this branch against main
- [X] All the tests are passing
- [X] My commit messages are descriptive and useful

### CODE REVIEW

File: src/test/java/sysc4806/group7/mini_shopify/MerchantControllerTest.java [hunk 1]
```diff
@@ -18,6 +18,10 @@ public class MerchantControllerTest {
     @Autowired
     private MockMvc mockMvc;
 
+    @Autowired
+    MerchantRepository merchantRepository;
+
+
```
Comment (ArthurAtangana) [inline src/test/java/sysc4806/group7/mini_shopify/MerchantControllerTest.java:24]: why is this needed?

Comment (Rebecca-Elliott) [reply src/test/java/sysc4806/group7/mini_shopify/MerchantControllerTest.java:24]: it's not. I was testing something and forgot to remove it.

**Review by ArthurAtangana — APPROVED** [review]

**Review by mpdesantis — APPROVED** [review]
Thanks for the video showing how this works. Looks great!
